Rogue-Ape-Studios/Crusader
# ✨Conditional Items

## Content

A very basic version of the item system, though it currently only supports lifesteal. Needs expansion in the future.

## Changes

### Added
`Assets/Prefabs/Items/BaseConditionalItem.prefab` : Was created to be the base template for all items.
`Assets/Scripts/Items/ConditionType.cs` : Was created for registering the correct item in the future.
`Assets/Scripts/Items/ConditionalItem.cs` : Was created for in-game use.
`Assets/Scripts/Player/Stats/RogueApeStudio.Crusader.Player.Stats.asmdef` : Was created to serve as the new namespace for the stats.

### Updated
`Assets/Prefabs/Items/BaseCommonItem.prefab` : Was renamed.
`Assets/Prefabs/Items/ItemManager.prefab` : Was updated with some basic functionality.
`Assets/Scripts/Health/Health.cs` : Was updated with some dependencies.
`Assets/Scripts/Health/RogueApe.Crusader.HealthSystem.asmdef` : Was updated with some dependecies.
`Assets/Scripts/Items/Item.cs` : Was updated, removed duplicate code.
`Assets/Scripts/Items/ItemManager.cs` : Was updated to support conditional items.
`Assets/Scripts/Items/RogueApeStudio.Crusader.Items.asmdef` : Was updated to reference basic attack and health.
`Assets/Scripts/Player/Combat/AttackDamage.cs` : Was updated to support conditional items.
`Assets/Scripts/Player/Combat/BasicAttack.cs` : Was updated to support backend changes to the stats system.
`Assets/Scripts/Player/Combat/RogueApeStudio.Crusader.Player.Combat.asmdef` : Was updated to include the stat dependency.
`Assets/Scripts/Player/Stats/CurrentPlayer.asset` : Was updated to be set back to default.
`Assets/Scripts/Player/Stats/PlayerScriptableObject.cs` : Was updated with a new namespace.

## Testing

The feature / Bugfix can be testing by following these steps:

1. Open a level.
2. Insert the item manager.
3. Make sure the item spawn is accessible to the player.
4. Play level.
5. Spawn item.
6. Collect item.
7. Player now has life steal
 
GIF posted in discord. Too large for github.


Closes #189
